server/publications/collections/product-publications.app-test.js

```diff
@@ -98,51 +132,71 @@ describe("Publication", function () {
         isSoldOut: false,
         isBackorder: false
       });
+      // a product for an inactive Merchant Shop
+      // this product is here to guard against false-positive test results
+      Collections.Products.insert({
+        ancestors: [],
+        title: "Lite Bright",
+        shopId: inactiveMerchantShopId,
+        type: "simple",
+        price: priceRangeA,
+        isVisible: true,
+        isLowQuantity: false,
+        isSoldOut: false,
+        isBackorder: false
+      });
+
+      // helper arrays for writing expecations in tests
+      merchantShop1ProductIds = [productId1, productId2, productId3];
+      merchantShop1VisibleProductIds = [productId2, productId3];
+      activeShopProductIds = [productId1, productId2, productId3, productId4, productId5];
+      activeShopVisibleProductIds = [productId2, productId3, productId4, productId5];
+      activeMerchantProductIds = [productId2, productId3, productId4];
 
       collector = new PublicationCollector({ userId: Random.id() });
     });
 
     describe("Products", function () {
-      it("should return all products to admins", function (done) {
```

Comment:
the diff is working a little too hard here. I replaced the "should return all products to admins" test with a few tests to get a little more granular